WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt

```diff
@@ -199,26 +201,49 @@ class EditPostPublishSettingsViewModel @Inject constructor(
         connections.isEmpty()
                 && publicizeTableWrapper.getServiceList().any { it.status != PublicizeService.Status.UNSUPPORTED }
 
-    private fun isPostPublished(): Boolean =
-        editPostRepository?.getPost()?.status?.let { post ->
-            post == PostStatus.PUBLISHED.toString()
-        } ?: false
-
     private fun updateInitialConnectionListSharingStatus() {
         if (isPostPublished()) {
             connections.map { it.isSharingEnabled = false }
-        } else {
-            // TODO refactor considering connection skip metadata
-            with(shareLimit) {
-                if (this is ShareLimit.Enabled) {
-                    connections.take(this.sharesRemaining).map {
+            return
+        }
+        val shareLimitValue = shareLimit
+        val skipConnections = postPublicizeSkipConnections()
+        if (shareLimitValue is ShareLimit.Enabled) {
+            // If shares remaining < number of connections, all connections are unchecked by default
+            if (shareLimitValue.sharesRemaining < connections.size) {
+                connections.map { it.isSharingEnabled = false }
+            } else {
+                run loop@{
+                    connections.take(shareLimitValue.sharesRemaining).map {
                         it.isSharingEnabled = true
                     }
+                    connections.forEachIndexed { index, connection ->
+                        // Use metadata to verify if this connection was previously disabled by the user
+                        skipConnections.firstOrNull { it.connectionId() == connection.connectionId.toString() }
+                            ?.let { connection.isSharingEnabled = it.isConnectionEnabled() }
+                    }
```

Comment:
Thanks for the changes @RenanLukas 🙇 . It seems the CI is complaining for the following reason

`EditPostPublishSettingsViewModel.kt:220:50 Parameter 'index' is never used, could be renamed to _`

Since we are there, I would maybe consider something like below, let me know wdyt 🙇 

```suggestion
               run {
                    connections.take(shareLimitValue.sharesRemaining).map {
                        it.isSharingEnabled = true
                    }
                    connections.forEach { connection ->
                        // Use metadata to verify if this connection was previously disabled by the user
                        skipConnections.firstOrNull { it.connectionId() == connection.connectionId.toString() }
                            ?.let { connection.isSharingEnabled = it.isConnectionEnabled() }
                    }
                }
```

Reply:
Thanks, @develric ! I've removed the only place `index` was being used and forgot to make it a regular `forEach` 🤦‍♂️